File: backend/app/schemas/notification.py
"""
app/schemas/notification.py

Schéma de sortie des notifications, enrichi avec le contexte entreprise
(pour répondre à "rappel pour quelle entreprise exactement") et des
schémas d'agrégats pour alimenter les graphiques de la page dédiée.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel


class NotificationOut(BaseModel):
    id: str              # identifiant synthétique, ex: "document_erreur-<uuid>"
    type: str
    message: str
    route: str            # route frontend vers laquelle naviguer au clic
    created_at: datetime
    entreprise_id: uuid.UUID | None = None
    entreprise_nom: str | None = None


class NotificationsOut(BaseModel):
    total: int
    notifications: list[NotificationOut]


class RepartitionParType(BaseModel):
    type: str
    total: int


class RepartitionParEntreprise(BaseModel):
    entreprise_id: uuid.UUID | None
    entreprise_nom: str
    total: int


class NotificationsStatsOut(BaseModel):
    par_type: list[RepartitionParType]
    par_entreprise: list[RepartitionParEntreprise]

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  type: "document_erreur" | "ecriture_anomalie" | "ecriture_a_verifier" | "document_nouveau";
  message: string;
  route: string;
  created_at: string;
  entreprise_id: string | null;
  entreprise_nom: string | null;
}

export interface NotificationsResponse {
  total: number;
  notifications: Notification[];
}

export interface RepartitionParType {
  type: string;
  total: number;
}

export interface RepartitionParEntreprise {
  entreprise_id: string | null;
  entreprise_nom: string;
  total: number;
}

export interface NotificationsStats {
  par_type: RepartitionParType[];
  par_entreprise: RepartitionParEntreprise[];
}